Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -631,13 +631,21 @@ void WebProcessProxy::addExistingWebPage(WebPageProxy& webPage, BeginsUsingDataS
     if (webPage.preferences().backgroundWebContentRunningBoardThrottlingEnabled())
         setRunningBoardThrottlingEnabled();
 #endif
-    if (!webPage.preferences().shouldTakeSuspendedAssertions())
-        m_throttler.setShouldTakeSuspendedAssertion(false);
-
     markProcessAsRecentlyUsed();
     m_pageMap.set(webPage.identifier(), &webPage);
     globalPageMap().set(webPage.identifier(), &webPage);
 
+    bool shouldTakeSuspendedAssertion = false;
+    for (auto& page : m_pageMap.values()) {
```

Comment:
The code duplication is very unfortunate. We should factor this out to a separate function to avoid it.
Maybe a `shouldTakeSuspendedAssertion()` function on WebProcessProxy.